packages/block-library/src/comment-template/index.php

```diff
@@ -8,11 +8,17 @@
 /**
  * Function that recursively renders a list of nested comments.
  *
- * @param WP_Comment[] $comments    The array of comments.
+ * @param WP_Comment[] $comments        The array of comments.
  * @param WP_Block     $block           Block instance.
+ * @param int          $depth           Depth of the current comment. We need to
+ * keep keep track of this to add the `depth-*` class to comments at different
+ * levels using `comment_class()`.
  * @return string
  */
-function block_core_comment_template_render_comments( $comments, $block ) {
+function block_core_comment_template_render_comments( $comments, $block, $depth = 1 ) {
+	global $comment_depth;
```

Comment:
I am not very happy that we have to use a global here (and reset it at the end of the function 🤦 ) but as far as I can tell there is no other way to add the correct `depth-*` class to the comment without it.

Reply:
Note that `$comment_depth` is an existing global in `get_comment_class()`, so there's probably not much harm in using it considering that some plugins might already rely on it being available: https://developer.wordpress.org/reference/functions/get_comment_class/

Reply:
gotcha, thanks @justintadlock ! 🙂